typing/typecore.ml

```diff
@@ -1922,17 +1984,12 @@ and type_pat_aux
   | Ppat_constraint(sp, sty) ->
       assert construction_not_used_in_counterexamples;
       (* Pretend separate = true *)
-      begin_def();
-      let cty, ty, force = Typetexp.transl_simple_type_delayed !env sty in
-      end_def();
-      generalize_structure ty;
-      let ty, expected_ty' = instance ty, ty in
-      unify_pat_types ~refine loc env ty (instance expected_ty);
+      let cty, ty, expected_ty' =
+        solve_Ppat_constraint ~refine loc env sty expected_ty in
       type_pat category sp expected_ty' (fun p ->
         (*Format.printf "%a@.%a@."
           Printtyp.raw_type_expr ty
           Printtyp.raw_type_expr p.pat_type;*)
-        pattern_force := force :: !pattern_force;
```

Comment:
do I understand correctly that setting `pattern_force` inside the `type_pat` continuation was an innocuous bug before, as the same forcing function could then be added several times to the queue due to backtracking?

Reply:
Since types generated by Parmatch do not contain type annotations, this would never occur, but the order was still strange.